docs/items/index.md

```diff
@@ -1,72 +1,127 @@
+- The stack size, typically between 1 and 64, obtainable through `itemstack.getCount()` and changeable through `itemstack.setCount(int)` or `itemstack.shrink(int)`.
+- The extra NBT data, where stack-specific data is stored. Obtainable through `itemstack.getTag()`, or alternatively through `itemstack.getOrCreateTag()` which accounts for no tag existing yet. A variety of other NBT-related methods exist as well, the most important being `hasTag()` and `setTag()`.
+
+To create a new `ItemStack`, call `new ItemStack(Item)`, passing in the backing item. By default, this uses a count of 1 and no NBT data; there are constructor overloads that accept a count and NBT data as well if needed.
+
+If you want to represent that a stack has no item, use `ItemStack.EMPTY`. If you want to check whether an `ItemStack` is empty, call `itemstack.isEmpty()`.
 
 ## Creative Tabs
 
-An item can be added to a `CreativeModeTab` via `BuildCreativeModeTabContentsEvent` on the [mod event bus][modbus]. An item(s) can be added without any additional configurations via `#accept`.
+By default, your item will only be available through `/give` and not appear in the creative inventory. Let's change that!
+
+The way you get your item into the creative menu depends on what tab you want to add it to.
+
+### Existing Creative Tabs
+
+An item can be added to an existing `CreativeModeTab` via the `BuildCreativeModeTabContentsEvent`, which is fired on the [mod event bus][modbus], only on the [logical client][sides]. Add items by calling `event#accept`.
 
 ```java
-// Registered on the MOD event bus
-// Assume we have RegistryObject<Item> and RegistryObject<Block> called ITEM and BLOCK
+//MyItemsClass.MY_ITEM is a Supplier<? extends Item>, MyBlocksClass.MY_BLOCK is a Supplier<? extends Block>
 @SubscribeEvent
-public void buildContents(BuildCreativeModeTabContentsEvent event) {
-  // Add to ingredients tab
-  if (event.getTabKey() == CreativeModeTabs.INGREDIENTS) {
-    event.accept(ITEM);
-    event.accept(BLOCK); // Takes in an ItemLike, assumes block has registered item
-  }
+public static void buildContents(BuildCreativeModeTabContentsEvent event) {
+    // Is this the tab we want to add to?
+    if (event.getTabKey() == CreativeModeTabs.INGREDIENTS) {
+        event.accept(MyItemsClass.MY_ITEM);
+        event.accept(MyBlocksClass.MY_BLOCK); // Accepts an ItemLike. This assumes that MY_BLOCK has a corresponding item.
+    }
 }
 ```
 
-You can also enable or disable items being added through a `FeatureFlag` in the `FeatureFlagSet` or a boolean determining whether the player has permissions to see operator creative tabs.
+The event also provides some extra information, such as `getFlags()` to get the list of enabled feature flags, or `hasPermissions()` to check if the player has permissions to view the operator items tab.
+
+:::note
+This method is for adding your items to Minecraft's tabs, or to other mods' tabs. To add items to your own tabs, see below.
+:::
```

Comment:
This should be at the top of the section and a link to the relevant section. Maybe the headers should also be changed to reflect this.